ConfigurationSystem/private/ConfigurationClient.py

```diff
@@ -141,7 +142,8 @@ def getOptionsDict(self, sectionPath):
     optionList = gConfigurationData.getOptionsFromCFG(sectionPath)
     if isinstance(optionList, list):
       for option in optionList:
-        optionsDict[option] = gConfigurationData.extractOptionFromCFG("%s/%s" % (sectionPath, option))
+        optionsDict[option] = gConfigurationData.extractOptionFromCFG("%s/%s" %
```

Comment:
Wouldn't this cause problems if one of the options in the optionList is a comma separated list of values and one of the entries contains __comma__?

Reply:
Well,  getOptionsDict() is a method for getting values without casting to any type. The resulting dictionary values are just strings and it is up to the client to interpret theses values. It can be that there are cases (that I do not know) where a list of values is extracted with getOptionsDict(). The values indeed will only contain real commas only, so we are simply back to the problem that we had before and simple List.fromChar() will split it badly. 
So, this is not a universal solution, of course. But practical problems like DNs with commas can be resolved without affecting other cases.